### PR TITLE
WE-767 Added MudLogView to frontend

### PR DESCRIPTION
**Pull request by KindCoder-no**
## Fixes
This pull request fixes WE-767

## Description
Added MudLogView to frontend

## Type of change

* New feature (non-breaking change which adds functionality)

## Impacted Areas in Application

* Frontend

## Checklist:

*Communication*
* [X] PR is related to an issue
* [ ] I have made corresponding changes to the documentation

*Code quality*
* [X] Code follows the style guidelines
* [X] I have self-reviewed my code
* [X] No new warnings are generated

*Test coverage*
* [x] Existing tests pass
* [ ] New code is covered by passing tests

### CODE REVIEW

File: Src/WitsmlExplorer.Frontend/components/ContentViews/MudLogsListView.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    if (selectedWellbore && selectedWellbore.mudLogs) {
+      setMudLogs(selectedWellbore.mudLogs);
+    }
+  }, [selectedWellbore]);
+
+  const getTableData = () => {
+    return mudLogs.map((msg, index) => {
+      return {
+        id: msg.uid,
+        index: index + 1,
```
Comment (janburak) [inline Src/WitsmlExplorer.Frontend/components/ContentViews/MudLogsListView.tsx:26]: index is unnecessary for MudLogs

File: Src/WitsmlExplorer.Frontend/models/mudLog.tsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import ObjectOnWellbore, { emptyObjectOnWellbore } from "./objectOnWellbore";
+
+// This is disabled because more attributes will be included later
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export default interface MudLogObject extends ObjectOnWellbore {}
```
Comment (janburak) [inline Src/WitsmlExplorer.Frontend/models/mudLog.tsx:5]: Can rename the interface to just MudLog

File: Src/WitsmlExplorer.Frontend/services/mudLogObjectService.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import MudLogObject, { emptyMudLogObject } from "../models/mudLog";
+import { Server } from "../models/server";
+import { ApiClient } from "./apiClient";
+
+export default class MudLogObjectService {
```
Comment (janburak) [inline Src/WitsmlExplorer.Frontend/services/mudLogObjectService.tsx:5]: Same here with removing "Object" from the name

Comment (janburak) [reply Src/WitsmlExplorer.Frontend/services/mudLogObjectService.tsx:5]: Also the file

File: Src/WitsmlExplorer.Frontend/services/mudLogObjectService.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  public static async getMudLogs(wellUid: string, wellboreUid: string, abortSignal?: AbortSignal): Promise<MudLogObject[]> {
+    const response = await ApiClient.get(`/api/wells/${wellUid}/wellbores/${wellboreUid}/mudlogs`, abortSignal);
+    if (response.ok) {
+      return response.json();
+    } else {
+      return [];
+    }
+  }
+
+  public static async getMudLogFromServer(wellUid: string, wellboreUid: string, uid: string, server: Server, abortSignal?: AbortSignal): Promise<MudLogObject> {
```
Comment (janburak) [inline Src/WitsmlExplorer.Frontend/services/mudLogObjectService.tsx:24]: Can remove this one as we have no plans of using it

File: Src/WitsmlExplorer.Frontend/components/ContentViews/MudLogsListView.tsx (hunk 1 above)
Comment (KindCoder-no) [reply Src/WitsmlExplorer.Frontend/components/ContentViews/MudLogsListView.tsx:26]: Fixed in WE-768

File: Src/WitsmlExplorer.Frontend/models/mudLog.tsx (hunk 2 above)
Comment (KindCoder-no) [reply Src/WitsmlExplorer.Frontend/models/mudLog.tsx:5]: Fixed in WE-768

File: Src/WitsmlExplorer.Frontend/services/mudLogObjectService.tsx (hunk 3 above)
Comment (KindCoder-no) [reply Src/WitsmlExplorer.Frontend/services/mudLogObjectService.tsx:5]: Fixed in WE-768

File: Src/WitsmlExplorer.Frontend/services/mudLogObjectService.tsx (hunk 4 above)
Comment (KindCoder-no) [reply Src/WitsmlExplorer.Frontend/services/mudLogObjectService.tsx:24]: Fixed in WE-768

**Review by janburak — APPROVED** [review]
- [X] Read issue and PR
- [X] Pulled branch and manually tested
- [X] Verify that PR resolves issue
- [X] Reviewed the code
Agreed that errors will be fixed in WE-768

**Pull request merged**